jobs/base_model_validator.py

```diff
@@ -115,6 +115,25 @@ def process(self, input_model):
             yield errors.ModelMutatedDuringJobError(model)
 
 
+class ValidateCommitType(beam.DoFn):
+    """DoFn to check whether commit type is valid"""
+
+    def process(self, input_model):
+        """Function that defines how to process each element in a pipeline of
+        models.
+
+        Args:
+            input_model: datastore_services.Model. Entity to validate.
+
+        Yields:
+            ModelCommitTypeError. Error for commit_type validation.
+        """
+        model = jobs_utils.clone_model(input_model)
```

Comment:
@brianrodri Why exactly do we need this? And what happens if we forget about this?

Reply:
We need to do this because Apache Beams demands that input values are not changed.

 Although validators don't tend to change the models, I'd rather enforce every kind of DoFn make a clone than have some jobs forget to.

Reply:
The thing I'm asking is if we should enforce it somewhere or move this to some kind of base class?

Reply:
I don't think we can move this into a base class. These classes need to be stateless, so we can't keep properties or stuff like that. The best we'd be able to do is change this to `self.model = super(...).process(model)`, or maybe `self.clone_model`, but that seems worse to me.